api-streaming/api-streaming.py

```diff
@@ -0,0 +1,143 @@
+
+        if os.path.exists(constants.UFM_API_SYSTEMS_RESULT):
+            stored_systems_api = read_json_from_file(constants.UFM_API_SYSTEMS_RESULT)
+
+        if os.path.exists(constants.UFM_API_PORTS_RESULT):
+            stored_ports_api = read_json_from_file(constants.UFM_API_PORTS_RESULT)
+
+        if os.path.exists(constants.UFM_API_LINKS_RESULT):
+            stored_links_api = read_json_from_file(constants.UFM_API_LINKS_RESULT)
+    except Exception as e:
+        logging.error(e)
+
+
+def update_ufm_apis():
+    global stored_versioning_api
+    global stored_systems_api
+    global stored_ports_api
+    global stored_links_api
+
+    try:
+        logging.info(f'Call update_ufm_apis')
+
+        ufm_new_version = send_ufm_request(constants.UFM_API_VERSIONING)
+
+        # check if systems api is changed
+        if (stored_versioning_api == '' or ufm_new_version["switches_version"] != stored_versioning_api[
+            "switches_version"] or
+                    ufm_new_version["switches_version"] != stored_versioning_api["switches_version"]):
+            stored_systems_api = send_ufm_request(constants.UFM_API_SYSTEMS)
+            write_json_to_file(constants.UFM_API_SYSTEMS_RESULT, stored_systems_api)
+
+        # check if ports api is changed
+        if stored_versioning_api == '' or ufm_new_version["ports_version"] != stored_versioning_api["ports_version"]:
+            stored_ports_api = send_ufm_request(constants.UFM_API_PORTS)
+            write_json_to_file(constants.UFM_API_PORTS_RESULT, stored_ports_api)
+
+        # check if links api is changed
+        if stored_versioning_api == '' or ufm_new_version["links_version"] != stored_versioning_api["links_version"]:
+            stored_links_api = send_ufm_request(constants.UFM_API_LINKS)
+            write_json_to_file(constants.UFM_API_LINKS_RESULT, stored_links_api)
+
+        stored_versioning_api = ufm_new_version
+    except Exception as e:
+        logging.error(e)
+
+
+def send_ufm_request(url):
+    ip = CONFIG['ufm-server-config']['ip']
+    username = CONFIG['ufm-server-config']['username']
+    password = CONFIG['ufm-server-config']['password']
+    ws_protocol = CONFIG['ufm-server-config']['ws_protocol']
+    url = ws_protocol + '://' + ip + '/ufmRest/' + url
+    headers = {}
+    # token auth: to be done
+    # if token:
+    #    headers = {"Authorization": "Bearer " + token['access_token']}
+    try:
+        logging.info(f'Send UFM API Request, URL: {url}')
+        response = requests.get(url, verify=False, headers=headers, auth=(username, password))
+        logging.info("UFM API Request Status [" + str(response.status_code) + "], URL " + url)
+        if response.raise_for_status():
+            logging.error(response.raise_for_status())
+    except Exception as e:
+        logging.error(e)
+    return response.json()
+
+# if run as main module
+if __name__ == "__main__":
```

Comment:
command line args needed, with help, verbose and args to override/provide cfg file, as here : https://docs.python.org/3/library/argparse.html